lancero_source.go

```diff
@@ -33,15 +33,15 @@ type LanceroSource struct {
 	ncards            int
 	clockMhz          int
 	active            []*LanceroDevice
-	chan2readoutOrder map[int]int
+	mixFraction       []float64
+	chan2readoutOrder []int // this could be a slice
```

Comment:
I made this a map instead of a slice on the theory that channel numbers might not be a contiguous set. I admit that this doesn't totally make sense. Also, I had the map backwards at first (that is, I mapped readout order to channel order, when it turned out we wanted the inverse map).

In other words, my thinking was muddled enough that I'll accept your idea to make this a slice instead of a map.